src/menu.cpp

```diff
@@ -2321,32 +2321,44 @@ void MainMenu::Draw( IMenuDrawer& menu_drawer, ITextDrawer& text_draw )
 
 MenuBase* MainMenu::ProcessEvent( const SystemEvent& event )
 {
-	if( event.type == SystemEvent::Type::Key &&
-		event.event.key.pressed )
-	{
-		if( event.event.key.key_code == KeyCode::Up )
-		{
-			PlayMenuSound( Sound::SoundId::MenuChange );
-			current_row_= ( current_row_ - 1 + 6 ) % 6;
+	SystemEvent::KeyEvent::KeyCode key = event.event.key.key_code;
 
-			if( current_row_ == 2 && !host_commands_.SaveAvailable() )
-				current_row_= ( current_row_ - 1 + 6 ) % 6;
-		}
+	switch(event.type)
+	{
+		case SystemEvent::Type::MouseKey:
+			if(event.event.mouse_key.mouse_button == SystemEvent::MouseKeyEvent::Button::Middle) key = KeyCode::Enter;
+		case SystemEvent::Type::Wheel:
+			if(event.event.wheel.dy > 0.1f) key = KeyCode::Up;
+			if(event.event.wheel.dy < 0.1f) key = KeyCode::Down;
```

Comment:
Let's convert wheel delta into some integer steps during SDL2 events processing and use here just `>0` and `<0`.